decode.go

```diff
@@ -174,7 +175,10 @@ func (d Decoder) decodeIntFromType(t Type, to reflect.Value) (err error) {
 			return
 		}
 
-		i, err = strconv.ParseInt(string(b), 10, 64)
+		i, err = strconv.ParseInt(unsafeString(b), 10, 64)
+		if err != nil {
+			_, err = strconv.ParseInt(string(b), 10, 64)
```

Comment:
Maybe leave a comment here explaining why we have to redo the operation to generate the error?